main.cpp

```diff
@@ -0,0 +1,29 @@
+#include "ip_filter.hpp"
+
+#include <algorithm>
+#include <cassert>
+#include <cstdlib>
+#if 0 /* Comlilers don't support it :/ */
+#include <execution>
+#endif
+#include <iostream>
+#include <string>
+#include <vector>
+
+int main(int, char const *[]) try
+{
+  std::vector<ip_filter::ip> ip_pool;
+
+  for(std::string line; std::getline(std::cin, line); )
+    ip_pool.emplace_back(line.substr(0, line.find('\t')));
+
+  std::sort(std::rbegin(ip_pool), std::rend(ip_pool));
```

Comment:
std::sort(ip_pool.begin(), ip_pool.end(), std::greater<ip_filter::ip>());

Reply:
Thanks! I'll fix it!